pytorch_lightning/plugins/training_type/ipu.py

```diff
@@ -113,7 +115,8 @@ def setup(self) -> None:
         # patch the dataloader creation function with the custom `poptorch.DataLoader`.
         # this violates the intended control flow for the plugins, but since this is experimental, we have chosen
         # to use the simpler solution before adding abstractions to override the `DataLoader` class
-        self.lightning_module.trainer._update_dataloader = self._convert_to_poptorch_loader
+        self.__update_dataloader_original = pytorch_lightning.utilities.data._update_dataloader
```

Comment:
I never use private attributes as I fear an user doing this:

```python
class IPUPlugin:
    def setup(self):
        self.__original = "foo"

    def teardown(self):
        print(self.__original)

    def __call__(self):
        self.setup()
        self.teardown()


class MyIPUPlugin(IPUPlugin):
    def teardown(self):
        print(self.__original)



IPUPlugin()()
MyIPUPlugin()()
```

`AttributeError: 'MyIPUPlugin' object has no attribute '_MyIPUPlugin__original'`